### PR TITLE
38 feature 38 create viewprogramsrecommendationsinterface use case layer and recommendationalgorithm class related interface

### PR DESCRIPTION
**Pull request by uyoyo0**

### CODE REVIEW

File: app/src/main/java/com/example/coursematchdaddy/clean_architecture_layers/presenters/classes/LoginPresenter.java
Comment (realdevvora) [inline app/src/main/java/com/example/coursematchdaddy/clean_architecture_layers/presenters/classes/LoginPresenter.java:1]: considering that these are blank methods, I'm assuming you are going to change them later right?

Comment (realdevvora) [reply app/src/main/java/com/example/coursematchdaddy/clean_architecture_layers/presenters/classes/LoginPresenter.java:1]: also be sure to add the changeView() method onto the doc

**Review by JaspreetKhela — CHANGES_REQUESTED** [review]
Please add comments to your code. Everything else looks good.

**Review by GagandeepLubana — APPROVED** [review]
LGTM

**Review by LVaishno — COMMENTED** [review]
The code makes sense and the comments look good

**Review by LVaishno — DISMISSED** [review]
Approved

**Review by JaspreetKhela — APPROVED** [review]

**Review by realdevvora — APPROVED** [review]

**Review by GagandeepLubana — APPROVED** [review]

**Pull request merged**